### PR TITLE
Refactor provider.getResourceUrl() to return a string instead of URL

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Changes the return value of `provider.getResourceUrl()` to return a string instead of a `new URL()`. It seems we only ever call `url.href()` within business logic to get the full string representation back. 

I discovered this while implementation `open-insights-provider-fastly` and found it a pain to map the string returned in from the server `SessionConfig` to a `new URL()`. I'm also concerned about the overhead of having to polyfill URL for older UAs in our provider. 

That being said, this isn't a strongly held opinion so happy to close this PR if y'all would prefer the URL type as I do see the benefits of runtime type checking.

### CODE REVIEW

**Review by jakewan — APPROVED** [review]
This makes sense to me. I tested it with the CBSi provider and the change is trivial.
